### PR TITLE
Review request for Microsoft.HybridCompute to add version preview/2024-05-20-preview

### PR DESCRIPTION
**Pull request by zhejiao**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Review by msft-aruk — APPROVED** [review]

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-05-20-preview/HybridCompute.json [hunk 1]
```diff
@@ -2862,6 +2862,26 @@
             "name": "PatchModeTypes",
             "modelAsString": true
           }
+        },
+        "enableHotpatching": {
+          "type": "boolean",
+          "description": "Captures the hotpatch capability enrollment intent of the customers, which enables customers to patch their Windows machines without requiring a reboot."
```
Comment (zhejiao) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-05-20-preview/HybridCompute.json:2868]: FYI, the CRP description is like: "Enables customers to patch their Azure VMs without requiring a reboot. For enableHotpatching, the 'provisionVMAgent' must be set to true and 'patchMode' must be set to 'AutomaticByPlatform'."

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-05-20-preview/HybridCompute.json [hunk 2]
```diff
@@ -4055,6 +4099,21 @@
       },
       "description": "Describes the properties of a Product License Profile ARM model."
     },
+    "HotpatchEnablementStatus": {
```
Comment (zhejiao) [inline specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-05-20-preview/HybridCompute.json:4102]: image

**Comment by zhejiao** [comment]
This PR adds a few properties based on the [latest version](https://github.com/Azure/azure-rest-api-specs/pull/28787) that was recently merged. There are no new breaking changes being introduced. The tag has been removed to allow the PR to proceed to review.

**Review by dkirby017 — APPROVED** [review]

**Comment by gary-x-li** [comment]
>           "type": "boolean",

ARM recommends enums over booleans for future proof APIs. Standard guidance is: replace boolean/switch properties with a more meaningful enum whenever possible.

A boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated].

---
Refers to: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-05-20-preview/HybridCompute.json:2867 in efcfad8. [](commit_id = efcfad8a0ee41310fca1fc8371e110e7a733635f, deletion_comment = False)

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-05-20-preview/HybridCompute.json (hunk 1 above)
Comment (gary-x-li) [reply specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-05-20-preview/HybridCompute.json:2868]: Any action needed for the comment above?

**Comment by zhejiao** [comment]
> > ```
> >       "type": "boolean",
> > ```
> 
> ARM recommends enums over booleans for future proof APIs. Standard guidance is: replace boolean/switch properties with a more meaningful enum whenever possible.
> 
> A boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated].
> 
> Refers to: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-05-20-preview/HybridCompute.json:2867 in [efcfad8](https://github.com/Azure/azure-rest-api-specs/commit/efcfad8a0ee41310fca1fc8371e110e7a733635f). [](commit_id = [efcfad8](https://github.com/Azure/azure-rest-api-specs/commit/efcfad8a0ee41310fca1fc8371e110e7a733635f), deletion_comment = False)

Hi gary, thanks for the suggestion. Based on this [link](https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md), this **_enableHotpatching_** field is just pure boolean type, which indicates the user's intent to enroll Hotpatch feature. It can only be True or False. Based on the guideline, keeping it as a boolean seems fine.

Also, we're adding this property in HybridCompute to mimic the existing behavior for Hotpatch enrollment in ComputeRP, as you can see [here](https://github.com/Azure/azure-rest-api-specs/blob/51031c3dc961c33be93afe1f15d35acfe5999861/specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-03-01/computeRPCommon.json#L1923C1-L1926C11).

File: specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-05-20-preview/HybridCompute.json (hunk 1 above)
Comment (zhejiao) [reply specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2024-05-20-preview/HybridCompute.json:2868]: No, this is just a reference (for reviewers) to the same property in the Compute RP. We don't need to do anything here.

**Pull request merged**